adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java

```diff
@@ -71,6 +77,25 @@ public void deleteAllDatasources() {
     StreamSupport.stream(allDatasourceConfigs.spliterator(), true).forEach(amqpPublisher::publishDeletion);
   }
 
+  public void validate(Long id) throws DatasourceNotFoundException, IOException, ValidationException {   
```

Comment:
Pls insert all relevant information as argument and return the status:
`public ValidationResult validate(Datasource datasource, Object importData)`

where 
`public class ValidationResult {
    health: ImportHealth; // enum with HEALTHY, WARNING, ERROR
    message: String; // => if warning or error add message here!
}

Reply:
Too late, I need to think about this one (maybe for an alternative approach) :D